weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/CallChat.tsx

```diff
@@ -0,0 +1,59 @@
+/**
+ * Code to determine if a call is a chat call, and if so,
+ * load the message data and render it.
+ */
+
+import _ from 'lodash';
+import React from 'react';
+
+import {LoadingDots} from '../../../../../LoadingDots';
+import {isWeaveRef} from '../../filters/common';
+import {ChatView} from '../PromptPage/ChatView';
+import {useCallAsChat} from '../PromptPage/hooks';
+import {TraceCallSchema} from '../wfReactInterface/traceServerClientTypes';
+import {CallSchema} from '../wfReactInterface/wfDataModelHooksInterface';
+
+type CallChatProps = {call: TraceCallSchema};
+
+// Does this look like a message object?
+export const isMessage = (message: any): boolean => {
+  if (isWeaveRef(message)) {
+    // Benefit of the doubt
+    return true;
```

Comment:
With the latest python client, we don't make these refs - maybe this should return false? I wonder what would happen if we think it is a message, but it is not in practice?

Reply:
Yeah, I left this in because Shawn's programmer projects were useful tests, will remove.

Reply:
i see now that you load the data later on and technically could re-check there as well. so up to you